xposed-bridge/src/main/java/de/robv/android/xposed/XSharedPreferences.java

```diff
@@ -480,25 +489,39 @@ public Editor edit() {
         throw new UnsupportedOperationException("read-only implementation");
     }
 
-    @Deprecated
+    /**
+     * Registers a callback to be invoked when a change happens to a preference file.<br>
+     * Note that it is not possible to determine which preference changed exactly and thus
+     * preference key in callback invocation will always be null.
+     *
+     * @param listener The callback that will run.
+     * @see #unregisterOnSharedPreferenceChangeListener
+     */
     @Override
     public void registerOnSharedPreferenceChangeListener(OnSharedPreferenceChangeListener listener) {
-        if (!mWatcherEnabled)
-            throw new UnsupportedOperationException("File watcher feature is disabled for this instance");
-
         synchronized(this) {
-            mListeners.put(listener, sContent);
+            if (listener != null && !mListeners.containsKey(listener)) {
+                mListeners.put(listener, sContent);
+                tryRegisterWatcher();
+            }
         }
     }
 
-    @Deprecated
+    /**
+     * Unregisters a previous callback.
+     *
+     * @param listener The callback that should be unregistered.
+     * @see #registerOnSharedPreferenceChangeListener
+     */
     @Override
     public void unregisterOnSharedPreferenceChangeListener(OnSharedPreferenceChangeListener listener) {
-        if (!mWatcherEnabled)
-            throw new UnsupportedOperationException("File watcher feature is disabled for this instance");
-
         synchronized(this) {
-            mListeners.remove(listener);
+            if (mListeners.containsKey(listener)) {
```

Comment:
HashMap remove returns a boolean that can be used here instead of containsKey

Reply:
remove(Object) is void. boolean return is only for remove(key, value). So I have to use .remove (listener, sContent)